android/cli/commands/_buildModule.js

```diff
@@ -293,6 +283,76 @@ AndroidModuleBuilder.prototype.initialize = function initialize(next) {
 	this.sharedHooksDir = path.resolve(this.projectDir, '..', 'hooks');
 
 	this.timoduleXmlFile = path.join(this.projectDir, 'timodule.xml');
+	this.timodule = fs.existsSync(this.timoduleXmlFile) ? new tiappxml(this.timoduleXmlFile) : undefined;
+	this.modulesDir = path.join(this.projectDir, 'modules', 'android');
+	this.globalModulesDir = path.join(this.globalModulesPath, 'android');
+
+	// process module dependencies
+	this.modules = this.timodule && !Array.isArray(this.timodule.modules) ? [] : this.timodule.modules.filter(function(m) {
+		if (!m.platform || /^android$/.test(m.platform)) {
+			var localPath = path.join(this.modulesDir, m.id),
+				globalPath = path.join(this.globalModulesDir, m.id),
+				getModulePath = function(modulePath) {
+					var items = fs.readdirSync(modulePath);
+					if (m.version) {
+						for (var item of items) {
+							if (item === m.version) {
+								m.path = path.join(modulePath, m.version);
+								return true;
+							}
+						}
+					} else if (items.length) {
+						var latest = items[items.length - 1];
+						if (!latest.startsWith('.')) {
+							m.version = latest;
+							m.path = path.join(modulePath, m.version);
+							return true;
+						}
+					}
+					return false;
+				}.bind(this);
+
+			if ((fs.existsSync(localPath) && getModulePath(localPath)) ||
+				(fs.existsSync(globalPath) && getModulePath(globalPath))) {
```

Comment:
I like to move longer if checks like this into a simple function with a fitting name so it's easier to understand what it does.